t/object-subscript.neon

```diff
@@ -0,0 +1,61 @@
+VAR a: Object := [1, 2, 3]
+VAR d: Object := {"one": 1, "two": 2, "three": 3}
+VAR x: Object := 5
+VAR n: Object := NIL
+
+TRY
+    print(n[0])
+TRAP DynamicConversionException AS dc DO
+    print(dc.info)
+END TRY
+%= object is null
+
+TRY
+    print(str(a[n]))
+TRAP DynamicConversionException AS dc DO
+    print(dc.info)
+END TRY
+%= index is null
+
+TRY
+    VAR y: Object := x[0]
+    print(str(y))
+TRAP ObjectSubscriptException DO
+    print("ObjectSubscriptException")
+END TRY
+%= ObjectSubscriptException
+
+TRY
+    print(str(a["42"]))
+TRAP DynamicConversionException AS dc DO
+    print(dc.info)
+END TRY
+%= to Number
+
+TRY
+    print(str(a[42]))
+TRAP ArrayIndexException AS ai DO
```

Comment:
Do not include a TRY/TRAP for exceptions that you do not expect. (Same with the following tests.) It obscures the actual test and makes it harder to read.